plugins/talk-plugin-notifications-category-moderation-actions/index.js

```diff
@@ -0,0 +1,144 @@
+const { get } = require('lodash');
+const path = require('path');
+
+const { MODERATION_NOTIFICATION_TYPES } = require('./config');
+
+const UNPUBLISHED_STATUS_TYPES = ['PREMOD', 'SYSTEM_WITHHELD'];
+
+const handle = async (ctx, comment) => {
+  const commentID = get(comment, 'id', null);
+  if (commentID === null) {
+    ctx.log.info('could not get comment id');
+    return;
+  }
+
+  // Check to see if this is a previously unpublished comment.
+  const commentHistory = get(comment, 'status_history', []);
+  // Check for last status before current one
+  const lastStatus = commentHistory[commentHistory.length - 2];
```

Comment:
This would error in the event that the status history is defined well.

Instead, perhaps walking backwards:

```suggestion
  let previouslyUnpublished = false;
  if (commentHistory.length >= 2) {
    for (let i = commentHistory.length - 2; i >= 0; i--) {
      const status = commentHistory[i];
      if (UNPUBLISHED_STATUS_TYPES.includes(status.type)) {
        previouslyUnpublished = true;
        break;
      }
    }
  }
```

It may need a bit of work!

Reply:
The issue with the proposed change is that in the event a comment was moderated, approved, and then flagged and approved again the commenter would receive 2 notifications (one upon initial publish and another when comment is approved again after the flag).

Reply:
@wyattjoh What do you mean by 'in the event that the status history is defined well'?

Reply:
That was a typo, my bad! I had meant:

> in the event that the status history is _not well defined_.

I'd just rather have defensive programming around accessing arrays, otherwise we could run into issues. Perhaps then this might work better?

```js
let previouslyUnpublished = false;
if (commentHistory.length >= 2) {
    const previousStatus = commentHistory[commentHistory.length - 2];
    if (UNPUBLISHED_STATUS_TYPES.includes(previousStatus.type)) {
        previouslyUnpublished = true;
    }
}
```

Reply:
Fine by me.

FWIW: by checking if true with (`lastStatus && lastStatus.type`), basically accomplishing the same thing already.